File: Desafios_Java/Ordenação_e_Filtros_em_Java/desafio_02.java
/* Desafio 2 - Compras no Supermercado */

package Desafios_Java.Ordenação_e_Filtros_em_Java;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Arrays;

public class desafio_02 {
    public static void main(String[] args) throws IOException {

        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        int nEntries = Integer.parseInt(reader.readLine());

        for (int counter = 0; counter < nEntries; counter++) {

            String[] list = reader.readLine().split(" ");

            Arrays.sort(list);

            int newLength = arrayFilter(list);

            for (int pos = 0; pos < newLength; pos++) {
                System.out.print(list[pos] + " ");
            }
            System.out.print("\n");
        }
    }

    private static int arrayFilter(String[] list) {
        String[] filter = new String[list.length];

        int index = 0;

        for (int i = 0; i < list.length - 1; i++) {
            if (!list[i].equals(list[i + 1])) {
                filter[index] = list[i];
                index++;
            }
        }
        filter[index++] = list[list.length - 1];

        for (int i = 0; i < index; i++) {
            list[i] = filter[i];
        }

        return index;
    }
}

File: Desafios_C_Sharp/Praticando_Programação_em_C_Sharp/desafio_02.cs
/* Desafio 2 - Compras no Supermercado*/

using System;

class desafio_02 {
  public static void Main (string[] args) {
    var totalDeCasosDeTeste = int.Parse(Console.ReadLine());
    
    for(int i = 0; i < totalDeCasosDeTeste; i++){
      
      string[] input = Console.ReadLine().Split(" ");
      
      Array.Sort(input);
      
      string[] list = new string[1];
      
      for(int j = 0; j < input.Length; j++){
        if(!Array.Exists(list, element => element == input[j])){
          list[list.Length-1] = input[j];
          Array.Resize(ref list, list.Length+1);
        }
      }

      foreach (string str in list){
        Console.Write(str+" ");
      }
      Console.WriteLine("");
    }
  }
}